modules/ipcscript.js

```diff
@@ -48,6 +50,21 @@ function IPCScript_getMetaStr() {
   return this._metaStr;
 });
 
+IPCScript.prototype.dedup = function() {
+  // scripts sent via e10s message manager contain new string instances every time a new one is created -> dedup them to reduce memory footprint
+
+  // alternative approaches:
+  // a) beam down IPC script instances only once via the parent process manager every time they change.
+  // b) creating resource URIs and loading user scripts straight from URIs might also be more elegant
+  //    see: https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/Using#Programmatically_adding_aliases
+  var old = instances.get(this.uuid);
+
+  if(old == this.textContent)
+    this.textContent = old
+  else
+    instances.set(this.uuid, this.textContent)
```

Comment:
Changes to scripts should always be notifying the Config.

But option A sounds a lot like "store the content of every script in memory forever" again, in a slightly different arrangement.  I'm curious what we can do with lazy getters and either sync messages or however that resource URI thing would work.